config.template.toml

```diff
@@ -165,11 +165,145 @@ model = "gpt-4o"
 #disable_vision = true
 
 [llm.gpt4o-mini]
+# Model to use
+model = "gpt-4o-mini"
 # API key to use
 api_key = "your-api-key"
 
-# Model to use
-model = "gpt-4o-mini"
+
+################################### Router ###################################
+# litellm Router configuration
+# https://litellm.vercel.app/docs/routing
+##############################################################################
+
+# The "router_config"section can only occur once in the config.toml file!
+#[router_config]
+
+# The default model to use for routing, which must be part of the "models" entries below.
+#default_model = "gpt-4o"
+
+# The strategy to use for routing (e.g., usage-based-routing, cost-based-routing)
+#routing_strategy = "usage-based-routing"
+
+# Number of retries to attempt when an operation fails
+#num_retries = 3
+
+# waits min. 5s before retrying request
+#retry_after = 15
+
+# cooldown model if it fails > 1 call in a minute
+#allowed_fails = 1
+
+# cooldown the deployment for 100 seconds if it num_fails > allowed_fails
+#cooldown_time = 100
+
+# raise timeout error if call takes > 240s
+#timeout = 240
+
+# Whether to cache responses for reuse
+#cache_responses = false
+
+# Additional cache configuration options, see documentation for details
+#[router_config.cache_kwargs]
+
+#[router_config.retry_policy]
+#exceptions_to_retry = [ "RateLimitError",]  # List of exceptions to retry on
+#max_retries = 8  # Maximum number of retries for the specified exceptions
+#retry_after = 15  # Time to wait before the first retry for the specified exceptions
+#retry_after_multiplier = 2  # Multiplier for exponential backoff calculation
+
+# --- Important Notes ---
+# - The model entries below are just examples, you can add, edit or remove these as needed.
+# - For each model entry, there is a corresponding set of "litellm_params" section,
+#   which must be specified in the exact same syntax as shown below.
+# - The syntax with the brackets [[ ]] is mandatory, even if there is only one model entry,
+#   or the whole config file may not be loaded!
+# - The model name must be unique in the list!
+# - The model name can be the same as the model alias.
+# - The empty lines are only added for readability, they are not required.
+# ----
+
+# --- 1st model example: Azure OpenAI
+
+# See: https://learn.microsoft.com/en-us/azure/ai-services/openai/quotas-limits
+# See: https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models
+
+#[[router_config.models]]
+# An alias name for this model, must be unique in the list
+#model_name = "gpt-4-vision-enhancements"
+
+# Each model can have its own set of parameters, note the single [] syntax!
+#[router_config.models.litellm_params]
+# The actual model name as per the provider's API
+#model = "azure/gpt-4-vision"
```

Comment:
This is actually not the model name, but the name of the user-defined deployment of the model. I'm going to remove this, at least because this kind of detail is always so, so confusing. We have tried to [document it here](https://docs.all-hands.dev/modules/usage/llms/azure-llms).

Reply:
Open the documentation link in the PR description... the Azure example is mentioned right there under "Load Balancing".

Reply:
I did, and in the python example code there's a comment saying it's actually the deployment 😭

Reply:
I mean this one, for example, https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models?tabs=python-secure . All the page talks about models, and then a tiny little comment says that model doesn't mean what we think it means! 😂